src/hotspot/share/runtime/threads.cpp

```diff
@@ -1336,10 +1336,7 @@ void Threads::print_on(outputStream* st, bool print_stacks,
   }
 
   PrintOnClosure cl(st);
-  cl.do_thread(VMThread::vm_thread());
-  Universe::heap()->gc_threads_do(&cl);
-  cl.do_thread(WatcherThread::watcher_thread());
-  cl.do_thread(AsyncLogWriter::instance());
+  non_java_threads_do(&cl);
```

Comment:
This certainly looks neater and simpler and ensures all NJT's will be processed. There is additional synchronization involved in using the `NonJavaThreads::Iterator` that I had to look closely at but I think it is safe to use in this context.